### PR TITLE
fix(step9): saturate Step9DreamingState.step_count at int32 max

### LINKED ISSUE #1905: steps: Step9DreamingState.step_count wraps negative at int32 max
## Summary

\`step9_update\` in \`alberta_framework/steps/step9.py\` advances the Step 9
guarded-dreaming facade's own lifetime clock with a bare int32 increment:

\`\`\`python
new_state = Step9DreamingState(
    ...
    step_count=state.step_count + 1,
)
\`\`\`

\`Step9DreamingState.step_count\` is a \`jnp.int32\` array (initialized via
\`jnp.array(0, dtype=jnp.int32)\` in \`init_step9_state\`). JAX/NumPy int32
addition wraps silently on overflow: at \`INT32_MAX\` (\`2**31 - 1\`), \`+ 1\`
produces \`INT32_MIN\` (\`-2**31\`), not a saturated or raised value.

This is the same lifetime-clock defect already fixed for
\`alberta_framework.streams.alberta_plan_step1\` (#1369),
\`alberta_framework.streams.feature_discovery\` (#1367),
\`alberta_framework.streams.out_of_class\` (#1434), and
\`alberta_framework.streams.synthetic\` (#1365) via
\`alberta_framework.core.normalizers._saturating_int32_counter_increment\`,
and matches the still-open siblings tracked in #1787 (MixedHorde), #1841
(UPGDLearner), and #1843 (GauntletStream).

## Impact

\`Step9DreamingState\` is part of the public Step 9 library surface (exported
from \`alberta_framework.steps\`). External callers that treat
\`state.step_count\` as a monotonically non-negative counter (checkpoint
cadence, logging, external telemetry) get a corrupted value after ~2.1
billion real transitions, with no error raised.

Note: this facade-level counter is separate from
\`model_state.step_count\` (the world model's own internal counter, gated at
\`config.dreaming_warmup_steps\`), which is a different field on a different
object and is not affected by this issue.

## Fix

Replace the bare \`+ 1\` with
\`alberta_framework.core.normalizers._saturating_int32_counter_increment\`,
matching the existing pattern in the streams modules above.

## Related, out of scope here

\`alberta_framework/steps/step7.py\`'s \`Step7DynaState.step_count\` has the
identical bare-\`+ 1\` pattern at its own \`step7_update\` return site. Left
unfixed in this issue/PR to keep the change to one file; worth a follow-up
issue if nobody has picked it up.

### PR DESCRIPTION
**Pull request by rama0x1**
Fixes #1905.

## What

`step9_update` in `alberta_framework/steps/step9.py` advanced the Step 9
guarded-dreaming facade's own lifetime clock with a bare
`state.step_count + 1` on an int32 array. JAX/NumPy int32 addition wraps
silently on overflow: at `INT32_MAX` (`2**31 - 1`), `+ 1` produces
`INT32_MIN` instead of saturating or raising.

`Step9DreamingState` is exported public library surface
(`alberta_framework.steps.Step9DreamingState`). External callers that treat
`state.step_count` as a monotonically non-negative counter (checkpoint
cadence, logging, telemetry) would see it silently go negative after ~2.1
billion real transitions.

This mirrors the already-fixed lifetime-clock defects in
`alberta_framework.streams.{alberta_plan_step1,feature_discovery,out_of_class,synthetic}`
(#1369, #1367, #1434, #1365) and the still-open siblings tracked for
`GauntletStream` (#1843), `UPGDLearner` (#1841), and `MixedHorde` (#1787).

Note: `model_state.step_count` (the world model's own internal counter,
already gated separately at `config.dreaming_warmup_steps`) is a different
field on a different object and was never affected by this bug — only the
Step 9 facade's own `state.step_count` was.

## Fix

Reuse the existing
`alberta_framework.core.normalizers._saturating_int32_counter_increment`
helper, matching the pattern already used in the streams modules listed
above.

## Deliberately out of scope

`alberta_framework/steps/step7.py`'s `Step7DynaState.step_count` has the
identical bare-`+ 1` pattern in `step7_update`. Left unfixed here to keep
this PR to one file/one variable per the repo's contribution norms; flagged
as a follow-up in the linked issue.

## Verification

Head SHA: `df7fe617e62d1810ebb27042de6e325e8345f979` (branch off
`origin/main` at `cc877f78566675214e2f356bd797f0f3c5ec1bb0`, no other
commits in between).

Failing-test-first: `tests/test_step9_lifetime_clock.py` was written and
confirmed to fail against the unpatched function (`git stash` on
`step9.py` reproduces `int(result.state.step_count) == -2147483648` instead
of `INT32_MAX`) before the one-line fix was applied.

```
.venv/bin/python -m pytest tests/test_step9_lifetime_clock.py -q -o addopts=""
3 passed

.venv/bin/python -m pytest tests/test_step9_production.py tests/test_step9_rng.py \
  tests/test_step9_lifetime_clock.py -q -o addopts=""
195 passed

.venv/bin/python -m pytest tests/test_pipeline.py tests/test_pipeline_hostile_validation.py \
  tests/test_step9_hostile_validation.py tests/test_steps_float32_validation_hostile.py \
  -q -o addopts=""
230 passed

.venv/bin/python -m pytest tests/test_normalizers.py tests/test_normalizers_hostile_validation.py \
  -q -o addopts=""
159 passed

.venv/bin/python -m ruff check alberta_framework/steps/step9.py tests/test_step9_lifetime_clock.py
All checks passed!

.venv/bin/python -m mypy
Success: no issues found in 190 source files

git diff --check
(clean)
```

This is a mechanism-level correctness fix (L0), not a benchmark result — no
`outputs/` artifact or evidence-registry claim is affected.

## Provenance

Client: `claude-code`, provider: `anthropic`, model: `claude-sonnet-5`,
lane: `rama0x1-asi-claude-worker`.

### CODE REVIEW

**Comment by lalalune** [comment]
Superseded by #1917, which rebases this fix onto current main, preserves the contributor commit, removes unrelated formatter churn where applicable, and validates the combined lifetime/timing panel.
